thoth/package_extract/image.py

```diff
@@ -546,6 +546,50 @@ def get_image_size(path: str) -> int:
     return total_size
 
 
+def _get_cuda_version(path: str) -> dict:
+    """Get the cuda version."""
+    res = dict()
+    # gathering version from version.txt file
+    version_path = path + "/usr/local/cuda/version.txt"
+    if os.path.isfile(version_path):
+        with open(version_path, "r") as f:
+            for line in f.readlines():
+                if line.startswith("CUDA Version"):
+                    res["/usr/local/cuda/version.txt"] = line[len("CUDA Version"):].strip()
+                    _LOGGER.info("CUDA version identifier was found in file %s", res["/usr/local/cuda/version.txt"])
+                    break
+                else:
+                    _LOGGER.debug("No CUDA version identifier was found in file %r", version_path[len(path):])
+    else:
+        _LOGGER.warning("No %r file was found", version_path[len(path):])
+
+    # gathering version from nvcc command
+    nvcc_path = glob.glob(path + '/usr/local/cuda/bin/nvcc')
+    if len(nvcc_path) > 0:
+        st = os.stat(nvcc_path[0])
+        os.chmod(nvcc_path[0], st.st_mode | stat.S_IEXEC)
+        result = run_command("{} --version".format(nvcc_path[0]), raise_on_error=False)
+        if result.return_code != 0:
+            _LOGGER.warning(
+                "Unable to detect CUDA version - nvcc returned non-zero version: %s",
+                result.to_dict(),
+            )
+        else:
+            for line in result.stdout.splitlines():
+                if line.startswith("Cuda compilation tools"):
+                    for version in line.split(", "):
+                        if version.startswith("V"):
+                            res["nvcc_version"] = version[1:]
+                            _LOGGER.info("Detected CUDA version from nvcc output: %s", res["nvcc_version"])
+                            break
+                        else:
+                            _LOGGER.debug("Unable to detect CUDA version from nvcc output: %r", result.stdout)
+    else:
+        _LOGGER.warning("No nvcc executable was found")
```

Comment:
Here just info - and why we were looking for nvcc (to detect CUDA version).

Reply:
Thanks, done!